### ISSUE TITLE
Unable to load ONNX model

### ISSUE THREAD

**Issue by nawinrajkumar** [opencv/opencv#24543]
### System Information

OpenCV version: 4.6.0
Operating System: ubuntu 20.04
GCC version: 9.4.0
Rust Binding Version: 0.86.1

### Detailed description

I have been trying to load an onnx model using **readNetFromONNX** and I get the following error.
![updated](https://github.com/twistedfall/opencv-rust/assets/83204531/4c2d7eb8-a6dd-4423-95c0-256891e36ec3)
I just want to know how to import the onnx model correctly into the function.

P.S: I used OpenCV Rust bindings for the code but the error is coming from the C++ header. 

### Steps to reproduce

`
let onnx_path = "/home/Projects/rust-inference-onnx/model.onnx";
let model: dnn::Net = dnn::read_net_from_onnx(onnx_path).expect("Unable to load model");
`

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by nawinrajkumar** [comment]
Yeah.. I accidentally took the opencv-python version though I'm not using python. I have looked into the issues and I don't seem to find an answer for that. Is my model path correct for yolov8
`
let onnx_path = "/home/Projects/rust-inference-onnx/model.onnx";
let model: dnn::Net = dnn::read_net_from_onnx(onnx_path).expect("Unable to load model");
`

**Comment by dkurt** [comment]
There is also a PR which demonstrates that at least one version of YOLOv8 works: https://github.com/opencv/opencv/pull/24396
